pkg/fanal/artifact/remote/git.go

```diff
@@ -34,12 +35,16 @@ func NewArtifact(rawurl string, c cache.ArtifactCache, artifactOpt artifact.Opti
 	if err != nil {
 		return nil, cleanup, err
 	}
+	println(artifactOpt.FetchBranch)
+	println(artifactOpt.FetchCommit)
+	println(artifactOpt.FetchTag)
+
+	println(tmpDir)
 
 	cloneOptions := git.CloneOptions{
 		URL:             u.String(),
 		Auth:            gitAuth(),
 		Progress:        os.Stdout,
-		Depth:           1,
```

Comment:
Can you tell me the reason you removed this line? I thought it would be helpful even when we fetch only a single branch.

Reply:
So this now, by default, stays removed because a user might pass the `--commit` argument for which we would need the entire commit history.

And for cases when its not a `--commit`, we set it back to `1` [here](https://github.com/aquasecurity/trivy/pull/2494/files#diff-1e3db99af526dcac318c7ae2da42c97f906ffbef448a9b8f34a6814f566f021fR56)

Reply:
It makes sense. Thanks.